From 84c750ace30c10b38755e3cff7468b7bcc788500 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Fri, 14 Mar 2025 11:19:29 -0500
Subject: [PATCH] meson: change default_library to default_both_libraries

default_both_libraries is the equivalent option to default_library, but
default_library only applies to `library()`, and not `both_libraries()`.

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/meson.build  | 2 +-
 test/meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index e3c518e3..756d898a 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -46,7 +46,7 @@ fcobjshash_h = custom_target('fcobjshash.h',
 # Define FcPublic appropriately for exports on windows
 fc_c_shared_args = []
 
-if host_machine.system() == 'windows' and get_option('default_library') in ['both', 'shared']
+if host_machine.system() == 'windows' and get_option('default_both_libraries') in ['both', 'shared']
   fc_c_shared_args += '-DFcPublic=__declspec(dllexport)'
   fc_c_shared_args += '-DDLL_EXPORT'
 endif
diff --git a/test/meson.build b/test/meson.build
index fa16a92e..1ae0da8d 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -18,7 +18,7 @@ if host_machine.system() != 'windows'
     ['test-crbug1004254.c', {'dependencies': dependency('threads')}], # for pthread
   ]
 
-  if get_option('default_library') == 'static'
+  if get_option('default_both_libraries') == 'static'
     tests += [
       ['test-issue110.c'],
       ['test-d1f48f11.c'],
-- 
2.48.1

